### PR TITLE
HttpCache: New test for revalidating responses with an expired TTL

### PR DESCRIPTION
**Pull request by mpdude**
| Q             | A
| ------------- | ---
| Branch?       | 2.7
| Bug fix?      | no
| New feature?  | no
| BC breaks?    | no
| Deprecations? | no
| Tests pass?   | yes
| Fixed tickets | 
| License       | MIT
| Doc PR        |

See #22035, in particular [this and the following comments](https://github.com/symfony/symfony/pull/22035#issuecomment-287572234).

### CODE REVIEW

File: src/Symfony/Component/HttpKernel/Tests/HttpCache/HttpCacheTest.php [hunk 1]
```diff
@@ -809,6 +809,40 @@ public function testValidatesCachedResponsesWithETagAndNoFreshnessInformation()
...
+        $this->request('GET', '/');
+        $this->assertHttpKernelIsNotCalled();
+        $this->assertEquals(200, $this->response->getStatusCode());
+        $this->assertEquals('Hello World', $this->response->getContent());
+        $this->assertTraceContains('fresh');
+
+        sleep(15); // expire the cache
+
+        $this->setNextResponse(304, array(), '', function (Request $request, Response $response) use ($time) {
+            $this->assertEquals($time->format(DATE_RFC2822), $request->headers->get('IF_MODIFIED_SINCE'));
```
Comment (fabpot) [inline src/Symfony/Component/HttpKernel/Tests/HttpCache/HttpCacheTest.php]: That does not work on PHP 5.3. You should create a`$that` var outside and pass it via `use`

Comment (mpdude) [inline src/Symfony/Component/HttpKernel/Tests/HttpCache/HttpCacheTest.php]: I always wonder whether you know this off by heart, or if the failing tests remind you of it?

Comment (fabpot) [inline src/Symfony/Component/HttpKernel/Tests/HttpCache/HttpCacheTest.php]: we all do the same mistake, so I don't need the test to fail to spot this kind of errors :)

Comment (mpdude) [inline src/Symfony/Component/HttpKernel/Tests/HttpCache/HttpCacheTest.php]: Fixed. Somehow I know this is going to happen to me again.

**Comment by fabpot** [comment]
Thank you @mpdude.

**Pull request closed without merging**